expat/tests/basic_tests.c

```diff
@@ -5272,17 +5273,18 @@ START_TEST(test_big_tokens_scale_linearly) {
       xml_failure(parser);
     }
 
-    assert_true(g_bytesScanned > approx_bytes); // or the counter isn't working
-    if (g_bytesScanned > max_scanned) {
+    assert_true(bytes_scanned > approx_bytes); // or the counter isn't working
+    if (bytes_scanned > max_scanned) {
       fprintf(
           stderr,
           "after all input: scanned=%u (factor ~%.2f) max_scanned: %u (factor ~%u)\n",
-          g_bytesScanned, g_bytesScanned / (double)approx_bytes, max_scanned,
+          bytes_scanned, bytes_scanned / (double)approx_bytes, max_scanned,
           max_factor);
       fail("scanned too many bytes");
     }
 
     XML_ParserFree(parser);
+    g_bytesScanned = NULL;
```

Comment:
Test failures will longjmp out of this function, so this line won't run. As a result, `g_bytesScanned` would still be pointing to the old stack location, which is likely to cause weird problems in remaining tests.

Maybe an alternative would be to just set g_bytesScanned in runtests.c, and let it be set for the whole test run?

Reply:
...or making sure to set `g_bytesScanned = NULL` somewhere in the test runner loop, e.g. the failure handler or even before every test.

Reply:
@Snild-Sony thanks for the review and sharing these findings! :+1: